### ISSUE TITLE
Errormessage

### ISSUE THREAD

**Issue by frag-p** [x74353/KeyJerk-Reactions#1]
Hello!
When using KeyJerk-Reactions while using PhotoBooth I get the following message in a PopUp Window:
_"The Audio Video Control menu bar item wasn't found. Make sure there is a running app that is actively using your camera."_

although the green video Icon is active kin the Menu bar and PhotoBooth opened!

Is there any solution?

**Comment by angeloruggieridj** [comment]
> Hello! When using KeyJerk-Reactions while using PhotoBooth I get the following message in a PopUp Window: _"The Audio Video Control menu bar item wasn't found. Make sure there is a running app that is actively using your camera."_
> 
> although the green video Icon is active kin the Menu bar and PhotoBooth opened!
> 
> Is there any solution?

When you open the green icon, do you see only the preview of your Facetime HD Camera or also the reactions panel?

**Comment by frag-p** [comment]
I get the following:

￼

**Comment by x74353** [comment]
@frag-p - For KeyJerk Reactions to work, the Reactions menu needs to be present. Like in the attached screen shot. Your most recent comment looks like you tried to add a screen shot but all I see is a [OBJ] icon. Maybe try reposting your screen shot?
![ScreenShot 2024-01-11 at 17 25 24](https://github.com/x74353/KeyJerk-Reactions/assets/1432223/b13e248b-2b9a-4cee-a9aa-9dc56744c1ab)

**Comment by frag-p** [comment]
uh, sorry about that - I hope it works this time. In addition I upload the screenshot of the error message:
![GREEN PANEL](https://github.com/x74353/KeyJerk-Reactions/assets/154758566/983ba5ed-bbab-496b-905b-64b09fc3f5d5)
![ERRORMESSAGE](https://github.com/x74353/KeyJerk-Reactions/assets/154758566/0dbb957b-74ed-486a-ba94-57bf88825003)

**Comment by x74353** [comment]
@kantmichel @frag-p The issue might be that English is not the language used for macOS. If so, should hopefully be a simple fix to the script.

**Comment by Sindbad5** [comment]
Tried it with MacOS "german setup".
I am not so firm with AppleScript and MacOS naming conventions. So it is not so easy to fix it.

In the english version: … if description of menuBarItem is "Audio and Video Controls" then …
In the german version: … if description of menuBarItem is "Audio- und Videosteuerung" then …
This works well for me.

Then there’s in the original version : … window "Control Center" of application process "Control Center" … 
I have to use at the end of the line  „ControlCenter“ without a blank character : … window "Control Center" of application process "ControlCenter" …

And I get an error message without knowing anything how to fix it: ... „window "Control Center" of application process "ControlCenter" couldn’t be read. I tried all english and german variations for „Control Center“ - no success.

Any idea where to find the right naming for the window?

Thanks!

**Comment by frag-p** [comment]
...might it be "Kontrollzentrum"?

**Comment by Sindbad5** [comment]
Nope ...
![Screnshot](https://github.com/x74353/KeyJerk-Reactions/assets/67185114/54f0c7f3-774c-4460-b28c-82a0f3caffde)

**Comment by x74353** [comment]
I had to run a script to return all of the elements to determine what that names were. Not 100% certain, but I modified something like this to get the desired target element names:

https://stackoverflow.com/questions/42231133/use-applescript-to-list-the-names-of-all-ui-elements-in-a-window-gui-scripting

**Comment by perenstrom** [comment]
I also ran into problems with the script. After listing the names I found that my window was called `Control Centre`, as in the UK spelling. So updating all `click UI element 2 of group 2 of group 1 of window "Control Center" of application process "ControlCenter"` to `click UI element 2 of group 2 of group 1 of window "Control Centre" of application process "ControlCenter"` worked for me.

To check your window name, add the following after the line `click menuBarItem`

```
tell application "System Events"
	tell its application process "ControlCenter"
		repeat with selectedWindow in every window
			log name of selectedWindow as string
		end repeat
	end tell
end tell
```

And @x74353, perhaps just change it to the following to skip the localisation bugs:
```
try
	tell application "System Events"
		click UI element 2 of group 1 of group 1 of window 0 of application process "ControlCenter"
	end tell
end try
```

**Comment by Sindbad5** [comment]
For me, the solution:
I had set "Automatically hide and hide menu bar" to "always" in the system settings.
If I set it to "Never" it works.

Is there a way to keep it "always" and activate it in the script if necessary?

**Comment by Sindbad5** [comment]
Another Question ...
When I am using Facetime ane OBS Studio at the same time and the ScreenShare- Mode (MacOS Sonoma) is activated it looks like in the Screenshots.

When Facetime is activated it works. 
When OBS is activated there is additionally the Screenshare-Feature activated and the script doesn't work anymore.

How to address the right "section" (the section 'Blackmagic Design' below 'Screenshare', where 'Reactions' are located)?

Thanks in forward.

![Screenshot](https://github.com/x74353/KeyJerk-Reactions/assets/67185114/6c6c03f2-0165-4424-91c2-58c9c7246283)

**Comment by x74353** [comment]
@frag-p @angeloruggieridj @perenstrom @kantmichel @Sindbad5 

I've just uploaded a dmg for KeyJerk Reactions 2. This new release includes initial support for non-English system languages. After the Welcome window is dismissed, KeyJerk Reactions will check whether your Mac's system language is set to English. If not, you will be prompted to define the spellings of the various UI elements that KeyJerk Reactions needs to click/interact with. Once that setup is complete, KeyJerk Reactions should be able trigger Reactions in any language. That said, I have only tested with German so far. 

If you've previously set KeyJerk Reactions to not show the Welcome window when launched, you probably will need to Reset Settings in KeyJerk Reactions' Settings window.

Also in this release is initial support for screen sharing mode.

The "Automatically show and hide menu bar" →  "Always" in system settings issue will likely persist, but I will look into whether a solution can be developed for that, though I am not too confident one is possible.
